sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -0,0 +1,506 @@
+   * @param schedule disbursement to be checked.
+   * @param disbursementValue award to be checked.
+   * @returns true if the funding should not be disbursed, otherwise, false.
+   */
+  private shouldStopFunding(
+    schedule: ECertDisbursementSchedule,
+    disbursementValue: DisbursementValue,
+  ): boolean {
+    return (
+      schedule.stopFullTimeBCFunding &&
+      schedule.studentAssessment.application.currentAssessment.offering
+        .offeringIntensity === OfferingIntensity.fullTime &&
+      BC_FUNDING_TYPES.includes(disbursementValue.valueType)
+    );
+  }
+
+  /**
+   * Get the list of the awards from all disbursements with the associated
+   * schedule. Useful to generate the overaward balance deduction record.
+   * @param disbursementSchedules schedules to have the awards listed.
+   * @param valueTypes types to be part of the final awards list.
+   * @returns awards filtered by the types requested with the associated
+   * disbursement schedule.
+   */
+  private getAwardsByTypeWithRelatedSchedule(
+    disbursementSchedules: ECertDisbursementSchedule[],
+    valueTypes: DisbursementValueType[],
+  ): AwardValueWithRelatedSchedule[] {
+    const result: AwardValueWithRelatedSchedule[] = [];
+    for (const relatedSchedule of disbursementSchedules) {
+      for (const awardValue of relatedSchedule.disbursementValues) {
+        if (valueTypes.includes(awardValue.valueType)) {
+          result.push({ relatedSchedule, awardValue });
+        }
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Try to deduct an overaward balance owed by the student due to some previous overaward in some
+   * previous application.
+   * @param awards specific loan award being adjusted (e.g CSLF, BCSL). This will contain one or
+   * two entries, always from the same award, from where the student debit will be deducted.
+   * The debit will try to be settle as much as possible with the first award. If it is not enough
+   * if will check for the second award (second disbursement), when present.
+   * The awards lists will be always from the same loan award code. For instance, the list list
+   * will contain one or two awards of type CSLF.
+   * @param overawardBalance total overaward balance be deducted.
+   */
+  private subtractOverawardBalance(
+    awards: DisbursementValue[],
+    overawardBalance: number,
+  ) {
+    let currentBalance = overawardBalance;
+    for (const award of awards) {
+      // Award amount that is available to be taken for the overaward balance adjustment.
+      const availableAwardValueAmount =
+        +award.valueAmount - +award.disbursedAmountSubtracted;
+      if (availableAwardValueAmount >= currentBalance) {
+        // Current disbursement value is enough to pay the debit.
+        // For instance:
+        // - Award: $1000
+        // - Overaward balance: $750
+        // Then
+        // - Award: $1000
+        // - overawardAmountSubtracted: $750
+        // - currentBalance: $0
```

Comment:
and effective amount is = 250

Reply:
Sorry, are you asking if the "effective amount is = 250"?
The effective amount is not calculated in this method.

Reply:
no..it not a commnet... I was processing